internal/webhooks/machine.go

```diff
@@ -173,6 +173,12 @@ func (webhook *Machine) validate(oldM, newM *clusterv1.Machine) error {
 }
 
 func isMachinePoolMachine(m *clusterv1.Machine) bool {
+	if m.Labels != nil {
+		if _, ok := m.Labels[clusterv1.MachinePoolNameLabel]; ok {
+			return true
```

Comment:
should this be an `and` instead of an `or`? i.e. the machine is only a machine pool if and only if it has the machine pool name label _and_  it has an owner ref of kind MachinePool?

Reply:
In other places, we treat it as a MachinePool Machine based on if it has the label or not as it was quicker. I can't recall why it was different here, but I decided to update it for consistency and used an `or` so it would not be more restrictive than the other checks. In general, though, I think we want to treat it as a MachinePool Machine based on the presence of the label alone as that's how we select Machines belonging to MachineSets or the control plane.